pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/PulsarLedgerManager.java

```diff
@@ -89,6 +89,20 @@ public LedgerMetadata deserialize(String path, byte[] content, Stat stat) throws
         store.registerListener(this::handleDataNotification);
     }
 
+    private static Throwable mapToBkException(Throwable ex) {
+        if (ex instanceof MetadataStoreException.NotFoundException) {
+            return BKException.create(BKException.Code.NoSuchLedgerExistsOnMetadataServerException);
```

Comment:
What about calling initCause and add the original error to the chain?

Reply:
fixed